File: SportifyApp/src/app/model/compte.ts
export class Compte {
  private _id: number | undefined;
  private _identifiant: string | undefined;
  private _mdp: string | undefined;
  private _mail: string | undefined;
  private _type: string | undefined;

  constructor(
    id?: number,
    identifiant?: string,
    mdp?: string,
    mail?: string,
    type?: string
  ) {
    this._id = id;
    this._identifiant = identifiant;
    this._mdp = mdp;
    this._mail = mail;
    this._type = type;
  }

  /**
   * Getter id
   * @return {number }
   */
  public get id(): number | undefined {
    return this._id;
  }

  /**
   * Getter identifiant
   * @return {string }
   */
  public get identifiant(): string | undefined {
    return this._identifiant;
  }

  /**
   * Getter mdp
   * @return {string }
   */
  public get mdp(): string | undefined {
    return this._mdp;
  }

  /**
   * Getter mail
   * @return {string }
   */
  public get mail(): string | undefined {
    return this._mail;
  }

  /**
   * Getter type
   * @return {string }
   */
  public get type(): string | undefined {
    return this._type;
  }

  /**
   * Setter id
   * @param {number } value
   */
  public set id(value: number | undefined) {
    this._id = value;
  }

  /**
   * Setter identifiant
   * @param {string } value
   */
  public set identifiant(value: string | undefined) {
    this._identifiant = value;
  }

  /**
   * Setter mdp
   * @param {string } value
   */
  public set mdp(value: string | undefined) {
    this._mdp = value;
  }

  /**
   * Setter mail
   * @param {string } value
   */
  public set mail(value: string | undefined) {
    this._mail = value;
  }

  /**
   * Setter type
   * @param {string } value
   */
  public set type(value: string | undefined) {
    this._type = value;
  }
}

File: AppliSportSpringBoot/src/main/java/projet/AppliSport/model/Compte.java
package projet.AppliSport.model;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;
import java.util.Objects;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;
import javax.persistence.SequenceGenerator;
import javax.persistence.Version;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotEmpty;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonSubTypes.Type;
import com.fasterxml.jackson.annotation.JsonTypeInfo;
import com.fasterxml.jackson.annotation.JsonView;

import projet.AppliSport.views.Views;

@Entity
@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
@SequenceGenerator(name = "seqCompte", sequenceName = "seq_compte", initialValue = 100, allocationSize = 1)
@JsonTypeInfo(use=JsonTypeInfo.Id.NAME,include=JsonTypeInfo.As.PROPERTY,property="type")
@JsonSubTypes({
	@Type(value=Admin.class,name="admin"),
	@Type(value=Club.class,name="club"),
	@Type(value=Utilisateur.class,name="utilisateur")
})
public abstract class Compte implements UserDetails {

	
	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seqCompte")
	@Column(name = "compte_appli")
	@JsonView(Views.Common.class)
	private Long id;
	
	@NotEmpty
	@Column(name = "compte_identifiant", length = 200, nullable = false)
	@JsonView(Views.Common.class)
	private String identifiant;
	
	@NotEmpty
	//@JsonView(Views.Common.class)
	@Column(name = "compte_mdp", length = 200, nullable = false)
	private String mdp;
	
	@Email
	@Column(name = "compte_mail", length = 200, nullable = false)
	@JsonView(Views.Common.class)
	private String mail;
	@Version
	private int version;
	
	public Compte() {
		
	}
	
	public Compte(String identifiant, String mdp, String mail) {
		this.identifiant = identifiant;
		this.mdp = mdp;
		this.mail = mail;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getIdentifiant() {
		return identifiant;
	}

	public void setIdentifiant(String identifiant) {
		this.identifiant = identifiant;
	}

	public String getMdp() {
		return mdp;
	}

	public void setMdp(String mdp) {
		this.mdp = mdp;
	}

	public String getMail() {
		return mail;
	}

	public void setMail(String mail) {
		this.mail = mail;
	}
	
	
	

	public int getVersion() {
		return version;
	}

	public void setVersion(int version) {
		this.version = version;
	}

	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Compte other = (Compte) obj;
		return Objects.equals(id, other.id);
	}

	

	
	
	
	
}
